db.py

```diff
@@ -1,82 +1,114 @@
 import os
-import sqlite3 as lite
-import config
-from collections import defaultdict
 import re
+import sqlite3 as lite
 
+import config
 
 class DbAccess:
     """ This provides access to the database to keep track of urls and views """
+
     def __init__(self, filename):
         """ Setup connection to file and create tables if they don't exist"""
         self.filename = filename
+        self.connection = None
 
         # Create folder for database file if it doesn't already exist
         if not os.path.exists(os.path.dirname(filename)):
             print('WARN: The parent directory for ' + filename + ' does not exist so it will be created.')
             os.makedirs(os.path.dirname(filename), exist_ok=True)
 
-        connection = lite.connect(filename)
-        connection.execute('pragma journal_mode=wal')
-        cursor = connection.cursor()
+        self.connect()
+        cursor = self.connection.cursor()
         cursor.execute('CREATE TABLE IF NOT EXISTS url (id INTEGER PRIMARY KEY, url VARCHAR(256), count INTEGER);')
 
+    def connect(self):
+        def regexp(expr, item):
+            reg = re.compile(expr)
+            return reg.search(item) is not None
+
+        self.connection = lite.connect(self.filename, check_same_thread=False)
```

Comment:
I have previously had issues with concurrency. There is a note on [stackoverflow here](https://stackoverflow.com/a/22739924) on why we shouldn't be using `check_same_thread`.

Reply:
I wasn't sure if opening a new connection for effectively every query yields better performance. But that post regarding concurrency seems to be plausible as well, so I reverted those changes.